lib/core/include/metadata.hpp

```diff
@@ -0,0 +1,223 @@
+        };
+
+        const std::map<std::string, entity_type> token_to_type {
+            {tokens::collection, entity_type::collection},
+            {tokens::object,     entity_type::object},
+            {tokens::user,       entity_type::user},
+            {tokens::resource,   entity_type::resource}
+        };
+
+        auto mod_avu_meta(rxComm_t& comm, const avu& _md, const avu& _dmd, entity_type _et, const std::string& _tgt, const std::string& _op)
+        {
+
+            modAVUMetadataInp_t inp{
+                const_cast<char*>(_op.c_str())
+              , const_cast<char*>(type_to_token.at(_et).c_str())
+              , const_cast<char*>(_tgt.c_str())
+              , const_cast<char*>(_md.attribute.c_str())
+              , const_cast<char*>(_md.value.c_str())
+              , const_cast<char*>(_md.units.c_str())
+              , const_cast<char*>(_dmd.attribute.c_str())
+              , const_cast<char*>(_dmd.value.c_str())
+              , const_cast<char*>(_dmd.units.c_str())
+              , nullptr // unused arg9
+            };
+
+            return rxModAVUMetadata(&comm, &inp);
+
+        } // mod_avu_meta
+
+    } // namespace
+
+
+    auto to_type(const std::string& _t)
+    {
+
+        if(token_to_type.find(_t) == token_to_type.end()) {
+            return entity_type::none;
+        }
+
+        return token_to_type.at(_t);
+
+    } // to_type
+
+
+    auto to_token(entity_type _t)
+    {
+
+        if(type_to_token.find(_t) == type_to_token.end()) {
+            return tokens::none;
+        }
+
+        return type_to_token.at(_t);
+
+    } // to_token
+
+
+    void set(rxComm_t& _comm, const avu& _md, entity_type _et, const std::string& _tgt)
+    {
+        if(auto ec = mod_avu_meta(_comm, _md, {}, _et, _tgt, "set"); ec < 0) {
+            THROW(ec, bf{"metadata set failed for [%s]"} % _tgt);
+        }
+
+    } // set
+
+
+    void add(rxComm_t& _comm, const avu& _md, entity_type _et, const std::string& _tgt)
+    {
+
+        if(auto ec = mod_avu_meta(_comm, _md, {}, _et, _tgt, "add"); ec < 0) {
+            THROW(ec, bf{"metadata add failed for [%s]"} % _tgt);
+        }
+
+    } // add
+
+
+    void remove(rxComm_t& _comm, const avu& _md, entity_type _et, const std::string& _tgt)
+    {
+
+        if(auto ec = mod_avu_meta(_comm, _md, {}, _et, _tgt, "rm"); ec < 0) {
+            THROW(ec, bf{"metadata remove failed for [%s]"} % _tgt);
+        }
+
+    } // remove
+
+
+    void modify(rxComm_t& _comm, const avu& _md, const avu& _dmd, entity_type _et, const std::string& _tgt)
+    {
+
+        avu tmp{
+              std::string{"n:"} + _dmd.attribute
+            , std::string{"v:"} + _dmd.value
+            , std::string{"u:"} + _dmd.units
+        };
+
+        if(auto ec = mod_avu_meta(_comm, _md, tmp, _et, _tgt, "mod"); ec < 0) {
+            THROW(ec, bf{"metadata modify failed for [%s]"} % _tgt);
+        }
+
+    } // modify
+
+
+    using avus = std::vector<avu>;
```

Comment:
is this a public type?

if not, it should be moved into the function that uses it.